packages/language-server/src/MessageHandler.ts

```diff
@@ -240,11 +241,22 @@ export function handleHoverRequest(
   return
 }
 
-/**
- *
- * This handler provides the initial list of the completion items.
- */
-export function handleCompletionRequest(
+function prismaFmtCompletions(
+  params: CompletionParams,
+  document: TextDocument,
+): CompletionList | undefined {
+  const text = document.getText(fullDocumentRange(document))
+
+  const completionList = textDocumentCompletion(text, params)
+
+  if (completionList.isIncomplete) {
+    return undefined
+  } else {
+    return completionList
```

Comment:
there are no other ways for `completionList` not to be undefined or unexpected besides `completionList.isIncomplete` being `true`?

Reply:
The only case where prisma-fmt will not return a valid completion list is when it panics (which I was careful to avoid, but of course it exercises the parser which is a big piece of code that will sometimes have bugs). You can see that in the signature for text_document_completion (called in https://github.com/prisma/prisma-engines/blob/main/prisma-fmt/src/lib.rs#L13). In case of error, it will return an empty completion list with `isIncomplete: true`.

In case of panics, you get an exception on the JS side, so I'm 100% sure that the completionList variable here is a completion list and not something else — with the current implementation of course.